london | may2025 | mohammad | aldali | form-controls

mohammadaldali opened the pull request:
<!--

You must title your PR like this:

london | 
may2025 | mohammad |
 aldali |
 form-controls

For example,

London | May-2025 | Carol Owen | Wireframe

Complete the task list below this message.
If your PR is rejected, check the task list.

-->

## Learners, PR Template

Self checklist

- [x] I have committed my files one by one, on purpose, and for a reason
- [x] I have titled my PR with REGION | COHORT_NAME | FIRST_NAME LAST_NAME | PROJ_NAME
- [x] I have tested my changes
- [x] My changes follow the [style guide](https://curriculum.codeyourfuture.io/guides/reviewing/style-guide/)
- [x] My changes meet the [requirements](./README.md) of this task



### Changelist

- Created a semantic HTML form for collecting t-shirt order info.
- Validated user name and email inputs.
- Restricted color selection to red, green, and blue.
- Restricted size selection to XS, S, M, L, XL, XXL.
- Removed CSS as per the spec.
- Removed unnecessary aria-label to clear W3C validation warning.
- Improved consistency in <option> values.


## Questions

Ask any questions you have for your reviewer.

cjyuan reviewed (commented):
A well-prepared PR makes it easier for reviewers to approve it with minimal back-and-forth.

The "PR Essentials" file on the [Canvas](https://codeyourfutur-yov6609.slack.com/archives/C07QX99JK7B) in the #cyf-code-review channel has some helpful tips to make your PRs more robust and ready for review.

Can you take a few minutes to check against the essentials, including how to prepare PR description? Doing so can help speed up the review process.

And as a practice to get better at using AI tools, try running your code by ChatGPT. You can ask ChatGPT
```
review HTML code:
...        <--- paste your HTML code here
```

Then take a look at the suggestions it gives you, and pick out the ones that actually make sense for your project. It's a great way to learn and improve your work.

Once you've finished making changes, please leave a comment on this PR so I’ll get notified. Thanks!

mohammadaldali commented:
hi #cjyuan
Thanks for the helpful suggestions!

I've reviewed the code with ChatGPT, cleaned up the structure, and added proper validations. Looking forward to your feedback!

cjyuan commented on Form-Controls/index.html:
```diff
@@ -4,24 +4,72 @@
+          <input type="email" id="email" name="user-email" placeholder="Enter your email" required>
+          <br>
+
+          <label for="password">Password:</label>
+          <span aria-label="required"><strong>*</strong></span>
+          <input type="password" id="password" name="password" placeholder="Enter your password" minlength="8" required>
```
It’s important to ensure the implementation adheres to the specifications outlined in the `README.md` to maintain consistency and avoid unexpected behavior. If you’d like to introduce new features you believe are useful, it’s best to request pre-approval first to ensure alignment with project goals.

cjyuan commented on Form-Controls/index.html:
```diff
@@ -4,24 +4,72 @@
+          <input type="password" id="password" name="password" placeholder="Enter your password" minlength="8" required>
+        </section>
+
+        <section>
+          <h2>Choose your T-shirt</h2>
```
For forms, using `<fieldset>` with `<legend>`  are specifically designed for grouping form controls and labeling them. As such, they are more semantic than using `<section>` with `<h2>`.

cjyuan commented on Form-Controls/index.html:
```diff
@@ -4,24 +4,72 @@
+
+        <section>
+          <h2>Choose your T-shirt</h2>
+          <p>Please fill all required information</p>
```
Why place this instruction in the middle of the form when all fields are required?

cjyuan commented on Form-Controls/index.html:
```diff
@@ -4,24 +4,72 @@
+        <section>
+          <label for="name">Your name:</label>
+          <span aria-label="required"><strong> * </strong></span>
+          <input type="text" id="name" name="user-name" placeholder="Enter your name" required>
```
Currently a user can enter a name consisting of only space characters (e.g., " "). Can you enforce a stricter validation rule using the `pattern` attribute to disallow any name that contains only space characters?

cjyuan reviewed (commented):
You are off to a solid start.

Can you address the following issues and the comments I left with the code to further improve your implementation?
1. While there is no error, W3C Markup Validation Service shows some warnings in your code. Can you address those warnings?

2. The Lighthouse accessibility score is high but not yet 100. Can you improve the score to 100?

3. In your PR description, in the "Changelist" section, can you briefly describe what you have implemented in this PR branch?

mohammadaldali commented:
Thanks for the feedback!

I’ve now addressed all the remaining accessibility issues:

- Removed unnecessary ARIA attributes.
- Added correct labels for all inputs and selects.
- Increased the size of inputs and buttons to meet the touch target size using minimal inline CSS.
- Re-tested with Lighthouse — Accessibility score is now 100%.

Please let me know if anything else needs changes. Thanks again!

cjyuan reviewed (commented):
1. There is an error in your code. Can you fix it?

2. I left a comment with the code. Can you respond to that comment?

Please note that in the CYF course, the recommended way to inform the reviewer of your changes is to do both of the following:
  - Reply to their feedback.
    - In the responses, clarify how each piece of feedback was addressed to demonstrate that you've carefully reviewed the suggestions.
    - Your response may trigger a notification (depending on the reviewer's settings), helping ensure they’re aware of the updates you’ve made. 
  - Replace the "Reviewed" label by a "Needs review" label (which you have done -- great!)

cjyuan commented:
Changes look great, but have you validated your code after the making changes to your code? There is always a chance that we may accidently introduce errors to our code when we edit our code.

mohammadaldali commented:
Hi,

Thank you for your feedback!

I reviewed my code and found some issues. I've made the necessary changes, and I believe it now meets all the requirements you mentioned.

Please let me know if anything else needs to be updated.

Thanks again!

cjyuan commented:
According to W3C Markup Validation Service, there are still errors in your HTML code.

mohammadaldali commented:
the warning shown in the W3C Validator is not related to the HTML code I wrote.
It appears to come from the original repository or external assets being loaded automatically.
Specifically, the warning is about a <link> element, which my code does not include at all.
I have not made any changes to the core str
ucture that could cause this issue.
<img width="1440" alt="‏لقطة الشاشة 2025-05-31 في 10 59 52" src="https://github.com/user-attachments/assets/95fb11b6-849b-44bf-92f2-bc27b980fed1" />

cjyuan commented:
The errors I see are these:

![image](https://github.com/user-attachments/assets/324f683d-0c74-44d8-adcf-8f8caec610bf)

cjyuan reviewed (commented):
According to https://validator.w3.org/, there is still an error in your code. Can you fix it?

You can use the above website to check for errors yourself.

cjyuan commented:
Changes look good.
